sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -1365,6 +1365,32 @@ trait QuaternaryLike[T <: TreeNode[T]] { self: TreeNode[T] =>
   protected def withNewChildrenInternal(newFirst: T, newSecond: T, newThird: T, newFourth: T): T
 }
 
+trait NaryLike[T <: TreeNode[T]] { self: TreeNode[T] =>
```

Comment:
what is the benefit of defining `NaryLike` over `TreeNode[T]`?

Reply:
Consistency with other `QuaternaryLike`, etc. which has been extended with expressions, plans, etc.